horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -982,4 +980,150 @@ public QueryResult queryDataSet(Integer datasetId, String jsonpath, boolean arra
       }
       return result;
    }
+
+   private void transform(Run run, String role) {
+      if (run == null || role == null) {
+         log.errorf("Transformation parameters error: run %s, role %s", run, role);;
+      }
+      Collection<RunService.RunSchema> runSchemas = getSchemas(run, role);
+      Test test = Test.findById(run.testid);
+      List<JsonNode> nodesToProcess = new ArrayList<JsonNode>();
+      boolean matched = false;
+
+      if (!runSchemas.isEmpty()){
+         for ( RunSchema s : runSchemas ){
+            if (hasMatch(s, run.data)) {
+               nodesToProcess.add(run.data);
+            }
+            int nodeSize = nodesToProcess.size();
+
+            if (nodeSize == 0) {
+               new DataSet(run.start, run.stop, "",
+                     test.id, run.data, run, 0, run.owner, run.access)
+                     .persist();
+            } else {
+               Collection<Transformer> transformers = test.transformers;
+               if (transformers.isEmpty()) {
+                  new DataSet(run.start, run.stop, "",
+                        test.id, run.data, run, 0, run.owner, run.access)
+                        .persist();
+               } else {
+                  for (Transformer t : transformers){
+                     StringBuilder array = new StringBuilder("[");
+                     boolean atStart = true;
+                     for (int nodePos = 0; nodePos <= nodeSize; nodePos += 1 ) {
+                        if (!atStart) {
+                           array.append(",");
+                           atStart = false;
+                        }
+                        JsonNode n = nodesToProcess.get(nodePos);
+                        if (!t.extractors.isEmpty()) {
+                           for ( NamedJsonPath e : t.extractors ){
+                              QueryResult result = queryData(run.id, e.jsonpath, s.uri, true);
```

Comment:
Ouch. I completely forgot when defining the `NamedJsonPath` that there's no later stage that would select the first match/all matches. My fault; your decision to use arrays here is sensible, though I will have to add the option later on.